sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpResponse.java

```diff
@@ -88,10 +72,30 @@ public final HttpRequest getRequest() {
         return request;
     }
 
+    /**
+     * Buffers the {@link HttpResponse}.
+     * <p>
+     * If {@link #isBuffered()} is true, this instance of {@link HttpResponse} is returned. Otherwise, a new instance
+     * of {@link HttpResponse} is returned where {@link #getBody()} is buffered into memory.
+     *
+     * @return The buffered {@link HttpResponse}.
+     */
+    public abstract HttpResponse buffer();
+
+    /**
+     * Whether this {@link HttpResponse} is buffered.
+     *
+     * @return Whether this {@link HttpResponse} is buffered.
+     */
+    public abstract boolean isBuffered();
+
     /**
      * Closes the response content stream, if any.
+     *
+     * @throws IOException If an I/O error occurs
      */
     @Override
-    public void close() {
+    public void close() throws IOException {
+        getBody().close();
```

Comment:
maybe

```suggestion
    public void close() {
        try {
            getBody().close();
        } catch (IOException ex) {
            throw LOGGER.logExceptionAsError(new UncheckedIOExpcetion(ex));
        }
```

so that we don't need to do it everywhere else?

Reply:
We should talk about this comment in the sync, this is being done for a larger reason.

Right now, we override a lot of interface methods that throw checked exceptions to not throw checked exceptions and in a lot of cases we need to do this try-catching within the implementation. This means we now are wrapping the underlying exception in another exception, which could already be wrapping an exception itself, resulting in the root cause being dozens of exception stack trace lines down. Or worse, there have been times where the caught exception gets dropped meaning the root cause is now lost.

So, I want to dive into this as a discussion point.